ts/Series/CompareCumulativeComposition.ts

```diff
@@ -0,0 +1,741 @@
+/* *
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import type DataExtremesObject from '../Core/Series/DataExtremesObject';
+
+import Axis from '../Core/Axis/Axis.js';
+import Point from '../Core/Series/Point.js';
+import Series from '../Core/Series/Series.js';
+
+const {
+    prototype: {
+        tooltipFormatter: pointTooltipFormatter
+    }
+} = Point;
+
+import U from '../Core/Utilities.js';
+const {
+    addEvent,
+    arrayMax,
+    arrayMin,
+    correctFloat,
+    defined,
+    isArray,
+    isNumber,
+    isString,
+    pick
+} = U;
+
+/* *
+ *
+ *  Declarations
+ *
+ * */
+
+declare module '../Core/Axis/AxisLike' {
+    interface AxisLike {
+        setCompare(compare?: string, redraw?: boolean): void;
+        setCumulative(cumulative?: boolean, redraw?: boolean): void;
+        setModifier(
+            mode: 'compare'|'cumulative',
+            modeState?: boolean|null|string,
+            redraw?: boolean
+        ): void;
+    }
+}
+
+declare module '../Core/Series/PointLike' {
+    interface PointLike {
+        change?: number;
+        cumulativeSum?: number;
+    }
+}
+
+declare module '../Core/Series/SeriesLike' {
+    interface SeriesLike {
+        compareValue?: number;
+        getCumulativeExtremes(activeYData: Array<number>): [number, number];
```

Comment:
Instead of extending `SeriesLike` with every method/property, perhaps it's better to store it all under one property?  What do you think @bre1470 ?

Like this:
```
interface ModifiersAdditions {
  compareValue?: number;
  getCumulativeExtremes(activeYData: Array<number>): [number, number];
  ...
}

declare module '../Core/Series/SeriesLike' {
    interface SeriesLike {
        modifiers: ModifiersAdditions
    }
}
```

Reply:
Just realized the [comment above](https://github.com/highcharts/highcharts/pull/16311#issuecomment-915977970) - okay!

Reply:
I agree.